src/Validation/Validator.php

```diff
@@ -556,9 +573,13 @@ protected function _processRules($field, ValidationSet $rules, $data, $newRecord
                 continue;
             }
 
-            $errors[$name] = $message;
             if (is_string($result)) {
                 $errors[$name] = $result;
+            } elseif ($this->_useI18n) {
+                $args = $rule->get('pass');
+                $errors[$name] = __d($this->_validationDomain, $name, $this->_translateArgs($args));
```

Comment:
Slightly OT: I guess you won't concur but what I don't like about the framework internally is that it loses a lot of natural english for its get text locales. Instead of __('First Name') it should use __('FIRST_NAME_LABEL') etc... or instead of __('Save!') it should use __('SAVE_BUTTON')...